### PR TITLE
implemented controller, dto, mapper, new exception, getBookById method

### PR DESCRIPTION
**Pull request by DmytroV95**

### CODE REVIEW

**Comment by SaiyaJ1N** [comment]
You have a great solution, let`s make it more better!

File: src/main/java/com/varukha/onlinebookstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.varukha.onlinebookstore.dto;
+
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+    private Long id;
```
Comment (SaiyaJ1N) [inline src/main/java/com/varukha/onlinebookstore/dto/CreateBookRequestDto.java:7]: Id field in CreateBookRequestDto should not be present

File: src/main/java/com/varukha/onlinebookstore/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
+        this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
```
Comment (SaiyaJ1N) [inline src/main/java/com/varukha/onlinebookstore/service/impl/BookServiceImpl.java:20]: We can use @RequiredArgsConstructor to automaticaly create and initialize all field

**Review by SaiyaJ1N — CHANGES_REQUESTED** [review]

**Review by SaiyaJ1N — APPROVED** [review]
Great!

File: src/main/java/com/varukha/onlinebookstore/repository/BookRepository.java [hunk 3]
```diff
@@ -2,9 +2,12 @@
...
 import java.util.List;
+import java.util.Optional;
 
 public interface BookRepository {
     Book save(Book book);
 
-    List<Book> findAll();
+    List<Book> getAll();
+
+    Optional<Book> findBookById(Long id);
```
Comment (vovasalyha) [inline src/main/java/com/varukha/onlinebookstore/repository/BookRepository.java:12]:
```suggestion
    Optional<Book> findById(Long id);
```

**Review by vovasalyha — APPROVED** [review]
GJ

**Pull request merged**